src/extension.ts

```diff
@@ -1,193 +1,330 @@
+      await this.handleActiveManifestChanged(manifest, isLastActive)
+    })
+    this.manifestManager.onDidRequestRebuild(async (manifest) => {
+      if (manifest === this.manifestManager.getActiveManifest()) {
+        console.log(`Manifest at ${manifest.uri.fsPath} requested a rebuild`)
+        await executeCommand(`${EXT_ID}.${TaskMode.clean}`)
+      }
+    })
 
-  console.log(`is VSCode running in sandbox: ${IS_SANDBOXED.toString()}`)
+    this.outputTerminal = new FlatpakTerminal()
+    this.extCtx.subscriptions.push(this.outputTerminal)
 
-  // Look for a flatpak manifest
-  const manifestFinder = new FlatpakManifestFinder()
-  const manifests = await manifestFinder.find()
+    this.runner = new FlatpakRunner(this.outputTerminal)
+    this.extCtx.subscriptions.push(this.runner)
+    this.runner.onDidFinishedTask(async (finishedTask) => {
+      await this.handleFinishedTask(finishedTask)
+    })
+  }
 
-  if (manifests.length > 0) {
-    // Make sures the documents portal is running
+  async activate() {
     await ensureDocumentsPortal()
+    await this.manifestManager.loadLastActiveManifest()
 
-    manifests.forEach((manifest) => store.manifestFound(manifest))
-    //TODO: allow the user to select a manifest
-    const manifest = manifests[0]
-    // Create the build directory if it doesn't exists
-    if (!(await exists(manifest.buildDir))) {
-      await fs.mkdir(manifest.buildDir)
-    }
-    // Mark the app as already initialized
-    store.manifestSelected(manifest)
-
-    const terminal = new FlatpakTerminal()
-    const runner = new FlatpakRunner(terminal)
-
-    window.registerTerminalProfileProvider(`${EXT_ID}.runtime-terminal-provider`, {
-      provideTerminalProfile: () => {
-        return new TerminalProfile(manifest.runtimeTerminal())
-      }
+    // Private commands
+    this.registerCommand('show-active-manifest', async () => {
+      await this.manifestManager.doWithActiveManifestElseDisplayErr(async (activeManifest) => {
+        await vscode.window.showTextDocument(activeManifest.uri)
+      })
     })
 
-    window.registerTerminalProfileProvider(`${EXT_ID}.build-terminal-provider`, {
-      provideTerminalProfile: () => {
-        return new TerminalProfile(manifest.buildTerminal())
-      }
+    // Public commands
+    this.registerCommand('select-manifest', async () => {
+      await this.manifestManager.selectManifest()
     })
 
-    context.subscriptions.push(
-      registerCommand(`${EXT_ID}.runtime-terminal`, () => {
-        const runtimeTerminal = window.createTerminal(manifest.runtimeTerminal())
+    this.registerCommand('runtime-terminal', async () => {
+      await this.manifestManager.doWithActiveManifestElseDisplayErr((activeManifest) => {
+        const runtimeTerminal = window.createTerminal(activeManifest.runtimeTerminal())
+        this.extCtx.subscriptions.push(runtimeTerminal)
+        this.activeTerminals.push(runtimeTerminal)
```

Comment:
ah yeah, that seems problematic. It is only used though to kill terminals when manifest changes. I think we can remove that since we can't also kill terminals that are provided by TerminalProviders.